pkg/client/cache/listers.go

```diff
@@ -77,37 +77,36 @@ type storePodsNamespacer struct {
 // the store; there may have been some filtering that already happened before
 // that.
 func (s storePodsNamespacer) List(selector labels.Selector) (pods api.PodList, err error) {
-	list := api.PodList{}
 	if s.namespace == api.NamespaceAll {
 		for _, m := range s.indexer.List() {
 			pod := m.(*api.Pod)
 			if selector.Matches(labels.Set(pod.Labels)) {
-				list.Items = append(list.Items, *pod)
+				pods.Items = append(pods.Items, *pod)
 			}
 		}
-		return list, nil
+		return
 	}
 
 	key := &api.Pod{ObjectMeta: api.ObjectMeta{Namespace: s.namespace}}
 	items, err := s.indexer.Index(NamespaceIndex, key)
 	if err != nil {
+		err = nil
```

Comment:
I am not sure this is optimal. At the very least, you should add a comment explaining why this needs to happen. Unfortunately named return values don't help much either.